### PR TITLE
Project Mongo API Amanda

### PR DESCRIPTION
**Pull request by amandaflod**
Happy Sunday!

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
+import booksData from './data/books.json'
```
Comment (carolinebirgersson) [inline server.js:7]: Good that you remember about one line break between outer and local dependencies!

File: server.js [hunk 2]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+}
+
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavailable at the moment" })
+  }
+})
+
```
Comment (carolinebirgersson) [inline server.js:49]: Great that you have an error message that is communicative!

File: server.js [hunk 3]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+app.get('/books', async (req, res) => {
+  const allBooks = await Book.find()
+  res.json(allBooks)
 })
 
-// Start the server
+app.get('/books/id/:bookID', async (req, res) => {
+  const bookByID = await Book.findOne({ bookID: req.params.bookID });
```
Comment (carolinebirgersson) [inline server.js:60]: Have in mind to use findById here instead of findOne :)

File: server.js [hunk 4]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+  if (bookByID) {
+    res.json(bookByID);
+  } else {
+    res.status(404).json({ error: 'Book not found' });
+  };
+});
```
Comment (carolinebirgersson) [inline server.js:66]: Consider catching the error by a Status 400 page.

File: server.js [hunk 5]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+
+app.get('/books/author/:author', async (req, res) => {
+  const author = req.params.author
+  const booksByAuthor = await Book.find({ authors: { "$regex": author, "$options": "i" } })
```
Comment (carolinebirgersson) [inline server.js:70]: Impressive that you use a regex! Nicely done 👌

File: server.js [hunk 6]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+  res.json(booksByAuthor)
+})
+
+app.get('/books/top-rated', async (req, res) => {
+  const topRatedBooks = await Book.find({ average_rating: { $gte: 4.5 } })
```
Comment (carolinebirgersson) [inline server.js:75]: I love that you choose to sort out the top-rated books. Good thought and very clean with the regex! 👏🏼

File: server.js [hunk 7]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+  if (topRatedBooks.length > 0) {
+    res.json(topRatedBooks)
+  } else {
+    res.status(404).json({ error: 'Book not found' })
+  }
+})
+
+app.get('/books/short', async (req, res) => {
+  const shortBooks = await Book.find({ num_pages: { $lt: 400 } });
```
Comment (carolinebirgersson) [inline server.js:84]: Same here! Impressed by all your regex solutions. Great job!

File: server.js [hunk 8]
```diff
@@ -3,36 +3,92 @@ import bodyParser from 'body-parser'
...
+  title: String,
+  authors: String,
+  average_rating: Number,
+  isbn: Number,
+  isbn13: Number,
+  language_code: String,
+  num_pages: Number,
+  ratings_count: Number,
+  text_reviews_count: Number
+});
```
Comment (carolinebirgersson) [inline server.js:29]: Clean and easy to read!

**Review by carolinebirgersson — COMMENTED** [review]
Great job Amanda! 👏🏼  I love your regex solutions and I'm feeling inspired to use more of them myself. I think your API works well and like all of the details you choose to lift out – short books, ratings, authors and id. If I would have to use this API in a frontend-project I'll be glad 😉

**Review by KarolCislowski — APPROVED** [review]
Great job here 💯

**Pull request closed without merging**